File: velhia.api/src/entities/match/match.ts
import { status } from "src/shared/enums/status";
import { winner } from "src/shared/enums/winners";
import Player from "./player";
import Play from "./play";
import MultiAgentSystem from "./multiAgentSystem";
import IMatch from "./iMatch";

export default class Match implements IMatch {
  public readonly begin: Date
  public end?: Date
  public time: number
  public readonly sa: Player
  public readonly mas: MultiAgentSystem
  public plays: Play[]
  public status: status
  public winner?: winner

  public constructor (
    _begin: Date,
    _end: Date,
    _time: number,
    _sa: Player,
    _mas: MultiAgentSystem,
    _plays: Play[],
    _status: status,
    _winner?: winner
  ) {
    this.begin = _begin
    this.end = _end
    this.time = _time
    this.sa = _sa
    this.mas = _mas
    this.plays = _plays
    this.status = _status
    this.winner = _winner
    Object.freeze(this)
  }
}

File: velhia.game/src/entities/match/match.py
from typing import TypedDict, List, Optional
from .mas import MultiAgentSystem
from .player import Player
from .play import Play


class Match(TypedDict):

    begin: str
    end: Optional[str]
    time: float
    sa: Player
    mas: MultiAgentSystem
    plays: List[Play]
    status: str
    winner: Optional[str]
